src/profile-logic/profile-data.js

```diff
@@ -2911,3 +2915,444 @@ export function hasThreadKeys(
+      ...oldNativeAllocations,
+      stack: oldNativeAllocations.stack.map(mapBacktraceSelfStack),
+    };
+  }
+
+  return { ...thread, samples, markers, jsAllocations, nativeAllocations };
+}
+
+/**
+ * Creates a new thread with modified frame and stack tables for "nudged" return addresses:
+ * All return addresses are moved backwards by one byte, to point into the "call"
+ * instruction. This allows symbolication to obtain accurate line numbers and inline frames
+ * for these frames.
+ * Addresses which were sampled from the instruction pointer register remain unchanged.
+ * Non-native frames (i.e profiler label frames or JS frames) also remain unchanged.
+ *
+ * This function is called at the end of profile processing. In the Gecko profile format,
+ * caller frame addresses are return addresses. In the processed profile format, caller
+ * frame addresses point at/into the call instruction.
+ *
+ * # Motivation
+ *
+ * When the profiler captures a stack, the frame addresses in the stack come from two
+ * different sources:
+ *
+ *  1. The top frame comes from the instruction pointer register. The register contains the
+ *     address of the instruction that is currently executing.
+ *  2. The other frames come from stack walking. During stack walking, we record the
+ *     return addresses for all caller functions that are on the stack. A "return address"
+ *     is the address that the CPU will jump to once the called function returns. It is the
+ *     address of the instruction *after* the "call" instruction.
+ *
+ * Here's an example, where _LZ4F_compressUpdate calls _LZ4F_localSaveDict and the
+ * CPU was observed executing an instruction in _LZ4F_localSaveDict.
+ * The frame addresses in the stack are: [..., 0x5783a, 0x57c17].
+ * (The example uses libmozglue.dylib 64EC2645330C3A0BA6E4EBCD28A1B5940.)
+ *
+ * Top of stack:
+ *   _LZ4F_localSaveDict:
+ *     57c10  push  rbp
+ *     57c11  mov   rbp, rsp
+ *     57c14  mov   rax, rdi
+ * --> 57c17  cmp   dword [rdi+0x20], 0x2    ; instruction pointer, address maps to line 808
+ *     57c1b  mov   rdi, qword [rdi+0xa8]
+ *     57c22  jle   loc_57c33
+ *     [...]
+ *
+ * Caller:
+ *   _LZ4F_compressUpdate:
+ *     [...]
+ *     5782d  jmp   loc_5765f
+ *     57832  mov   rdi, rbx
+ *     57835  call  _LZ4F_localSaveDict       ; call instruction, address maps to line 897
+ * --> 5783a  test  eax, eax                  ; return address,   address maps to line 898
+ *     5783c  mov   rcx, 0xffffffffffffffff
+ *     [...]
+ *
+ * Corresponding C code:
+ *
+ * [...]
+ * 893    if ((cctxPtr->prefs.frameInfo.blockMode==LZ4F_blockLinked) && (lastBlockCompressed==fromSrcBuffer)) {
+ * 894        if (compressOptionsPtr->stableSrc) {
+ * 895            cctxPtr->tmpIn = cctxPtr->tmpBuff;
+ * 896        } else {
+ * 897            int const realDictSize = LZ4F_localSaveDict(cctxPtr);   // <-- here is the call
+ * 898            if (realDictSize==0) return err0r(LZ4F_ERROR_GENERIC);
```

Comment:
```suggestion
 * 898            if (realDictSize==0) return err0r(LZ4F_ERROR_GENERIC); // <-- here is the line pointed by the return address
```

Reply:
I used `here is the line we'd get for the return address`.